R/splatPop-simulate.R

```diff
@@ -0,0 +1,849 @@
+                                       params=params,
+                                       gff=gff,
+                                       key=key,
+                                       verbose=verbose)
+    
+    sim_sc <- splatPopSimulateSC(sim_means=sim_means$means, 
+                                 params=params, 
+                                 method=method,
+                                 counts_only=counts_only, 
+                                 verbose=verbose)
+    
+    out <- list(key=sim_means$key, means=sim_means$means, sc=sim_sc)
+    
+    return(out)
+}
+
+
+#' splatPopSimulateMeans
+#'
+#' Simulate mean expression levels for all genes for all samples, with between 
+#' sample correlation structure simulated with eQTL effects and with the option
+#' to simulate multiple groups (i.e. cell-types).
+#'
+#' @param vcf vcfR object containing genotypes of samples to simulate.
+#' @param params splatPopParams object containing parameters for population
+#'        scale simulations. See \code{\link{splatPopParams}} for details. 
+#'        Default=`newSplatPopParams()`.
+#' @param gff Either FALSE (if random.genes=TRUE) or a data.frame object 
+#'        containing a GFF/GTF file.
+#' @param key Either FALSE or a data.frame object containing a full or partial
+#'        splatPop key.
+#' @param verbose logical. Whether to print progress messages. Default=TRUE.
+#' @param ... any additional parameter settings to override what is provided in
+#'        \code{params}.
+#'
+#' @details splatPopParams can be set in a variety of ways. 1. If
+#' not provided, default parameters are used. 2. Default parameters can be
+#' overridden by supplying desired parameters using \code{\link{setParams}}.
+#' 3. Parameters can be estimated from real data of your choice using
+#' \code{\link{splatPopEstimate}}.
+#'
+#' `splatPopSimulateMeans` involves the following steps:
+#' \enumerate{
+#'     \item Load population key or generate random or GFF/GTF based key.
+#'     \item Format and subset genotype data from the VCF file.
+#'     \item If not in key, assign expression mean and variance to each gene.
+#'     \item If not in key, assign eGenes-eSNPs pairs and effect sizes.
+#'     \item If not in key and groups >1, assign subset of eQTL associations as
+#'     group-specific and assign DEG group effects.
+#'     \item Simulate mean gene expression matrix without eQTL effects
+#'     \item Quantile normalize by sample to fit single-cell expression
+#'     distribution as defined in `splatEstimate`.
+#'     \item Add quantile normalized gene mean and cv info the eQTL key.
+#'     \item Add eQTL effects to means matrix.}
+#'
+#' @return A list containing: `means` a data.frame (or list of dataframes if
+#' n.groups > 1) with the simulated mean gene expression value for each gene
+#' (row) and each sample (column) and `key` a data.frame with population 
+#' information including eQTL and group effects.
```

Comment:
I wonder if we can store these in `SplatPopParams` and return that? This is what I did for `kersplat` but it was a bit fiddly to get it working...

Reply:
Done!